File: Stage24/Java/Problem11401.java
package Stage24.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Problem11401 {
    static final long MOD = 1000000007;
    static long[] fact;

    static long modPow(long base, long exp) {
        long result = 1;
        while (exp > 0) {
            if ((exp & 1) == 1) result = result * base % MOD;
            base = base * base % MOD;
            exp >>= 1;
        }
        return result;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        fact = new long[N + 1];
        fact[0] = 1;

        for (int i = 1; i <= N; i++) {
            fact[i] = fact[i - 1] * i % MOD;
        }

        long numerator = fact[N];
        long denominator = fact[K] * fact[N - K] % MOD;

        long answer = numerator * modPow(denominator, MOD - 2) % MOD;
        System.out.println(answer);
        br.close();
    }
}

File: Stage24/C/Problem11401.c
#include <stdio.h>

#define MOD 1000000007LL

long long fact[4000001];

long long modPow(long long base, long long exp) {
    long long result = 1;
    while (exp > 0) {
        if (exp % 2 == 1) result = result * base % MOD;
        base = base * base % MOD;
        exp /= 2;
    }
    return result;
}

int main(void) {
    int N, K;
    scanf("%d %d", &N, &K);

    fact[0] = 1;
    for (int i = 1; i <= N; i++) {
        fact[i] = fact[i - 1] * i % MOD;
    }

    long long numerator = fact[N];
    long long denominator = fact[K] * fact[N - K] % MOD;

    long long answer = numerator * modPow(denominator, MOD - 2) % MOD;
    printf("%lld\n", answer);

    return 0;
}